### PR TITLE
phpstorm 2024.1.5

### PR DESCRIPTION
**Pull request by ghost**
After deleting phpstorm I noticed that Trash did not contained any related files and folders, checked deeper, they added `JetBrains` on top of couple directories and change some preferences file paths.

---

### CODE REVIEW

**Comment by ivan-kovnatskyi** [comment]
I'm sorry I'm running nix-homebrew, so brew path is not writable so commands would not run:

```console
❯ brew audit --cask --online phpstorm
Warning: audit is a developer command, so Homebrew's
developer mode has been automatically turned on.
To turn developer mode off, run:
  brew developer off

Error: Permission denied @ rb_sysopen - /nix/store/v9hchqmfnyy61kizif3s2z1dp0kqp25n-brew-4.3.1-patched/Library/Homebrew/vendor/bundle/ruby/3.3.0/.homebrew_vendor_version
homebrew-cask on  master 
❯ brew style --fix phpstorm
Error: Permission denied @ rb_sysopen - /nix/store/v9hchqmfnyy61kizif3s2z1dp0kqp25n-brew-4.3.1-patched/Library/Homebrew/vendor/bundle/ruby/3.3.0/.homebrew_vendor_version
```

But I think the change should be good anyway.

**Comment by khipp** [comment]
@ivan-kovnatskyi The elements of the `zap` stanza have to be sorted alphabetically (see [CI failure](https://github.com/Homebrew/homebrew-cask/actions/runs/10252272454/job/28362978643?pr=181602#step:9:25)).

File: Casks/p/phpstorm.rb [hunk 1]
```diff
@@ -26,10 +26,11 @@
...
-    "~/Library/Caches/PhpStorm#{version.major_minor}",
-    "~/Library/Logs/PhpStorm#{version.major_minor}",
-    "~/Library/Preferences/jetbrains.phpstorm.*.plist",
-    "~/Library/Preferences/PhpStorm#{version.major_minor}",
+    "~/Library/Application Support/JetBrains/PhpStorm#{version.major_minor}",
+    "~/Library/Application Support/JetBrains/consentOptions",
+    "~/Library/Caches/JetBrains/PhpStorm#{version.major_minor}",
+    "~/Library/Logs/JetBrains/PhpStorm#{version.major_minor}",
+    "~/Library/Preferences/com.jetbrains.PhpStorm.plist",
+    "~/Library/Preferences/jetbrains.jetprofile.asset.plist",
```
Comment (khipp) [inline Casks/p/phpstorm.rb:34]:
```suggestion
    "~/Library/Application Support/JetBrains/consentOptions",
    "~/Library/Application Support/JetBrains/PhpStorm#{version.major_minor}",
    "~/Library/Caches/JetBrains/PhpStorm#{version.major_minor}",
    "~/Library/Logs/JetBrains/PhpStorm#{version.major_minor}",
    "~/Library/Preferences/com.jetbrains.PhpStorm.plist",
    "~/Library/Preferences/jetbrains.jetprofile.asset.plist",
```

**Comment by krehel** [comment]
> I'm sorry I'm running nix-homebrew, so brew paths are not writable so commands would not run:
> 
> ```
> ❯ brew audit --cask --online phpstorm
> Warning: audit is a developer command, so Homebrew's
> developer mode has been automatically turned on.
> To turn developer mode off, run:
>   brew developer off
> 
> Error: Permission denied @ rb_sysopen - /nix/store/v9hchqmfnyy61kizif3s2z1dp0kqp25n-brew-4.3.1-patched/Library/Homebrew/vendor/bundle/ruby/3.3.0/.homebrew_vendor_version
> homebrew-cask on  master 
> ❯ brew style --fix phpstorm
> Error: Permission denied @ rb_sysopen - /nix/store/v9hchqmfnyy61kizif3s2z1dp0kqp25n-brew-4.3.1-patched/Library/Homebrew/vendor/bundle/ruby/3.3.0/.homebrew_vendor_version
> ```
> 
> But I think the change should be good anyway.

`nix-homebrew` is not one of our projects.  This won't be merged until the CI failures are addressed.

**Review by khipp — APPROVED** [review]
Thank you for your first contribution to `homebrew-cask`, @ivan-kovnatskyi! 🎉

**Comment by khipp** [comment]
It seems like upstream has published a new release. We also have to update the `version` and `sha256` stanzas:

```ruby
  version "2024.1.5,241.18968.24"
  sha256 arm:   "80d618b4518b48459c05f0470f8721439bf4bcb4e88f221ed1ffa2319bc02627",
         intel: "25314e9b6b84ab1c3f7bbcd004a80746c14c2005dcd2298f703479080339b97c"
```

**Review by khipp — APPROVED** [review]

**Pull request merged**